### PR TITLE
CLI flag for docker create(run) to change block device size.

### PR DESCRIPTION
**Pull request by shishir-a412ed**
Signed-off-by: Shishir Mahajan <shishir.mahajan@redhat.com>

### CODE REVIEW

**Comment by shishir-a412ed** [comment]
ping @rhatdan

File: daemon/graphdriver/devmapper/deviceset.go [hunk 1]
```diff
@@ -1732,10 +1736,74 @@ func (devices *DeviceSet) AddDevice(hash, baseHash string) error {
...
+	info.lock.Lock()
+	defer info.lock.Unlock()
+
+	if err := devices.activateDeviceIfNeeded(info, false); err != nil {
+		return fmt.Errorf("Error activating devmapper device for '%s': %s", hash, err)
+	}
+
+	defer devices.deactivateDevice(info)
+
+	fstype, err := ProbeFsType(info.DevName())
```
Comment (runcom) [inline daemon/graphdriver/devmapper/deviceset.go]: this is now stored in `devices.BaseDeviceFilesystem` (need latest docker/docker master code)

**Comment by rhvgoyal** [comment]
Not sure if we can use --storage-opt for this. tibor had mentioned that --storage-opt is daemon flag and should not be used in docker-run.

**Comment by rhvgoyal** [comment]
I guess for now continue with --storage-opt. It kind of makes sense to me. During review if somebody has better ideas we will change it.

**Comment by rhatdan** [comment]
Yes lets let docker make that call.  storage-opt makes most sense to me.

File: daemon/graphdriver/devmapper/deviceset.go (hunk 1 above)
Comment (shishir-a412ed) [inline daemon/graphdriver/devmapper/deviceset.go]: Fixed.

**Comment by shishir-a412ed** [comment]
@rhvgoyal No I have not looked into the comments from @rhatdan 
I am just trying to rebase it once a day. As rebase for this PR is complex, as it breaks almost everytime I rebase it with `upstream/master`.

I don't see the comments anymore (they have somehow magically disappeared :) ). I ll try to check if I have them on my github e-mail notifications.

Better way would be to do the review comments on line number in the code (the way @runcom did).
You can just click on the `+` sign before the line number and comment over there. That is easier for me to track as well.

I am traveling this week. I ll be back home on 21-Dec-2015. Will try to take a look then.

**Comment by runcom** [comment]
> I don't see the comments anymore (they have somehow magically disappeared :) ). I ll try to check if I have them on my github e-mail notifications.

they disappeared because you force push after a rebase and github lose tracks of old commit sha(s) :(

**Comment by shishir-a412ed** [comment]
@runcom 
ahh I see :) ... I believe I have to force push, to keep the same commit.
Or is there a better way to do it ?

**Comment by runcom** [comment]
> @runcom 
> ahh I see :) ... I believe I have to force push, to keep the same commit.
> Or is there a better way to do it ?

I guess you don't have option other than force pushing after a rebase :) as you suggested it's better to comment directly on the PR rather than on the single commit

**Comment by rhvgoyal** [comment]
If it is code comment, then it has to be on the commit. So I would think that you can rebase the patch and keep it internally and don't force push. Force push only after you have taken care of previous round of review comments.

**Comment by rhatdan** [comment]
Commented from other pull request.

Should storage-opt be added to the driver struct rather then being passed in at the create?

**Comment by shishir-a412ed** [comment]
@rhvgoyal @rhatdan I have fixed the earlier comments.
Throwing an error for other graph drivers is pending. Need to check on that.

**Comment by shishir-a412ed** [comment]
Closing this one. PR opened upstream.
https://github.com/docker/docker/pull/19123

**Pull request closed without merging**